### PR TITLE
feat(ui): cache package repositories (#763)

### LINKED ISSUE #763: Cache `PackageRepository` custom resources
**Is your feature request related to a problem? Please describe.**
When turning on logging, I noticed that we already have some client-side throttling again because of requests to the `PackageRepository` resources:

```
Installing kubernetes-dashboard...
I0606 13:42:59.093760  206148 request.go:629] Waited for 180.502406ms due to client-side throttling, not priority and fairness, request: GET:https://192.168.58.2:8443/apis/packages.glasskube.dev/v1alpha1/packagerepositories/glasskube
I0606 13:42:59.293507  206148 request.go:629] Waited for 359.303205ms due to client-side throttling, not priority and fairness, request: GET:https://192.168.58.2:8443/apis/packages.glasskube.dev/v1alpha1/packagerepositories
I0606 13:42:59.493378  206148 request.go:629] Waited for 392.101617ms due to client-side throttling, not priority and fairness, request: GET:https://192.168.58.2:8443/apis/packages.glasskube.dev/v1alpha1/packagerepositories
I0606 13:42:59.693269  206148 request.go:629] Waited for 397.435373ms due to client-side throttling, not priority and fairness, request: GET:https://192.168.58.2:8443/apis/packages.glasskube.dev/v1alpha1/packagerepositories/glasskube
I0606 13:42:59.893667  206148 request.go:629] Waited for 397.891212ms due to client-side throttling, not priority and fairness, request: GET:https://192.168.58.2:8443/apis/packages.glasskube.dev/v1alpha1/packagerepositories/glasskube
```

**Describe the solution you'd like**
* Investigate whether it is reasonable that so so many requests are being made (probably yes)
* Implement the same kind of cache as we have for `Package` and `PackageInfo` resources

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by christophenne**
Closes #763

## ✅ Checks
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -16,6 +16,9 @@ issues:
       linters:
         - dupl
         - lll
+    - path: "pkg/client/(.+)_cache.go"
```
Comment (christophenne) [inline .golangci.yml:19]: The code of the three caches is too duplicate to pass the check, but to be honest I didn't find a proper abstraction that doesn't also make it less readable. But I am open to inputs on this.

Comment (kosmoz) [reply .golangci.yml:19]: I created https://github.com/glasskube/glasskube/pull/777 to address this. Let me know what you think :slightly_smiling_face:

Comment (christophenne) [reply .golangci.yml]: thanks, LGTM :+1:

**Comment by christophenne** [comment]
closed in favor of #791

**Pull request closed without merging**